docs/custom_node_development.md

```diff
@@ -0,0 +1,126 @@
+linked statically or dynamically. OpenCV is a built in component in OVMS which could be used to perform manipulation on the image
+data. 
+
+Below are explained the data structures and functions defined in the API header. 
+
+### "CustomNodeTensor" struct 
+
+The CustomNodeTensor struct consist of several fields defining the data in the output and input of the node execution.
+Because it has constant length, it can carry multiple inputs. Custom node can generate results based on multiple inputs 
+from one or more other nodes. Each input can be referenced using an index:
+```
+inputTensor0 = &(inputs[0])
+```
+Every CustomNodeTensor struct include the following fields:
+`const char* name`  - pointer to the string representing the input name
+`uint8_t* data` - pointer to data buffer. Data is stored as bytes.
+`uint64_t dataBytes` - the size of the data allocation in bytes
+`uint64_t* dims` - pointer to the buffer storing array shape size. Size of each dimension consumes 8bytes
+`uint64_t dimsCount` - number of dimension on the data array
+`CustomNodeTensorPrecision precision` - data precision selector
+
+### CustomNodeTensorInfo struct
+
+The fields in struct CustomNodeTensorInfo are similar like in CustomNodeTensor. It just holds information about 
+the metadata of the custom node interfaces: inputs and outputs.
+
+### "CustomNodeParam" struct
+
+Struct CustomNodeParam stores a list of pairs with pointers to parameter and and value strings.
+Each parameter can be references in such object using an index.
+
+### "execute" function
+```
+int execute(const struct CustomNodeTensor* inputs, int inputsCount, struct CustomNodeTensor** outputs, int* outputsCount, const struct CustomNodeParam* params, int paramsCount);
+```
+
+This function implements the data transformation of the custom node. The input data for the function are passed in the form of 
+a pointer to CustomNodeTensor struct object. It includes all the data and pointers to buffers for all custom node inputs.
+The parameter inputsCount pass info about the number of inputs passed that way.
+
+Note that the execute function should not modify the buffers storing the input data because that would alter the data
+which potentially might be used in other pipeline nodes.
+
+The behaviour of custom node execute function can be dependent on the node parameters set in OVMS configuration.
+They are passed to execute function in `params` argument. paramsCount pass info about the number of parameters configured.
+
+The results of the data transformation should be returned by the outputs pointer to a pointer which stores the address of 
+CustomNodeTensor struct. Number of outputs is to be defined during the function execution in the `outputsCount` argument.
+
+Note that during the function execution all the output data buffers needs to be allocated. They will be released by OVMS after 
+the request processing is completed and returned to the user. The cleanup is triggered by calling `release` function.
```

Comment:
by calling `release` function....which also needs to be implemented in custom library.